File: channel-info/test_yt.js
const fs = require('fs');

async function main() {
    const channel = process.argv[2] || 'mkbhd';
    const url = `https://www.youtube.com/@${channel}`;
    try {
        const response = await fetch(url, {
            headers: {
                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
                'Accept-Language': 'en-US,en;q=0.9'
            }
        });
        const html = await response.text();
        fs.writeFileSync('yt.html', html);
        
        const match = html.match(/var ytInitialData = (\{.*?});<\/script>/);
        if (match) {
            fs.writeFileSync('yt.json', match[1]);
            console.log('Saved yt.json');
        } else {
            console.log('No ytInitialData found');
            console.log('HTML start:', html.substring(0, 500));
        }
    } catch (e) {
        console.error(e);
    }
}
main();

File: channel-info/test_yt.py
import urllib.request
import re
import json
import sys

def test():
    channel = sys.argv[1] if len(sys.argv) > 1 else "mkbhd"
    url = f"https://www.youtube.com/@{channel}"
    print(f"Fetching {url}...")
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0','Accept-Language': 'en-US,en;q=0.9'})
    html = urllib.request.urlopen(req).read().decode('utf-8')

    match = re.search(r'var ytInitialData = (\{.*?\});</script>', html)
    if match:
        data = json.loads(match.group(1))
        with open('yt.json', 'w', encoding='utf-8') as f:
            json.dump(data, f, indent=2)
        print("Saved yt.json")
    else:
        print("No ytInitialData found")

if __name__ == "__main__":
    test()
